src/librustc_typeck/check/mod.rs

```diff
@@ -3355,8 +3355,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         if let Some(suggested_field_name) =
                             Self::suggest_field_name(def.non_enum_variant(),
                                                      &field.as_str(), vec![]) {
-                                err.span_label(field.span,
-                                               format!("did you mean `{}`?", suggested_field_name));
+                                err.span_suggestion_with_applicability(
+                                    field.span,
+                                    "did you mean",
+                                    suggested_field_name.to_string(),
+                                    Applicability::MachineApplicable,
```

Comment:
Sorry, but I don't think this should be labelled as `MachineApplicable`. I suggest going with `MaybeIncorrect`.

It is machine applicable in the sense that this single suggestion can be automatically changed. But it is not guaranteed to actually fix the problem and make the code compile; it is only a suggestions. At least that what I get from reading this -- correct me if I'm wrong this will always work!

Reply:
This is something I've been unclear on when writing structured suggestions -- I think I've gotten conflicting answers! Someone should clarify in the docs for `Applicability`.

Reply:
Fixed both instances.